client/rest-high-level/src/main/java/org/elasticsearch/client/core/TermVectorsResponse.java

```diff
@@ -0,0 +1,452 @@
+            return tv;
+        }
+
+        public String getFieldName() {
+            return fieldName;
+        }
+
+        public void setFieldName(String fieldName) {
+            this.fieldName = fieldName;
+        }
+
+        public List<Term> getTerms() {
+            return terms;
+        }
+
+        public void setTerms(List<Term> terms) {
+            this.terms = terms;
+            if (terms != null) {
+                Collections.sort(this.terms, Comparator.comparing(Term::getTerm));
+            }
+        }
+
+        public FieldStatistics getFieldStatistics() {
+            return fieldStatistics;
+        }
+
+        public void setFieldStatistics(FieldStatistics fieldStatistics) {
+            this.fieldStatistics = fieldStatistics;
+        }
+
+        @Override
+        public boolean equals(Object obj) {
+            if (this == obj) return true;
+            if (!(obj instanceof TermVector)) return false;
+            TermVector other = (TermVector) obj;
+            return fieldName.equals(other.fieldName)
+                && Objects.equals(fieldStatistics, other.fieldStatistics)
+                && Objects.equals(terms, other.terms);
+        }
+
+        @Override
+        public int hashCode() {
+            return Objects.hash(fieldName, fieldStatistics, terms);
+        }
+
+        public static final class FieldStatistics {
+            private final long sumDocFreq;
+            private final int docCount;
+            private final long sumTotalTermFreq;
+
+            public FieldStatistics(long sumDocFreq, int docCount, long sumTotalTermFreq) {
+                this.sumDocFreq = sumDocFreq;
+                this.docCount = docCount;
+                this.sumTotalTermFreq = sumTotalTermFreq;
+            }
+            public int getDocCount() {
+                return docCount;
+            }
+            public long getSumDocFreq() {
+                return sumDocFreq;
+            }
+            public long getSumTotalTermFreq() {
+                return sumTotalTermFreq;
+            }
+            @Override
+            public boolean equals(Object obj) {
+                if (this == obj) return true;
+                if (!(obj instanceof FieldStatistics)) return false;
+                FieldStatistics other = (FieldStatistics) obj;
+                return docCount == other.docCount
+                    && sumDocFreq == other.sumDocFreq
+                    && sumTotalTermFreq == other.sumTotalTermFreq;
+            }
+
+            @Override
+            public int hashCode() {
+                return Objects.hash(docCount, sumDocFreq, sumTotalTermFreq);
+            }
+        }
+
+        public static final class Term {
+            private String term;
+            private int termFreq = -1;
+            private int docFreq = -1;
+            private long totalTermFreq = -1;
+            private float score = -1f;
+            private List<Token> tokens = null;
+
+            public Term() {};
```

Comment:
Why is this one mutable when the others are immutable? Maybe make all immutable?